// simple-react-calendar/lib/calendar/day.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classnames = _interopRequireDefault(require("classnames"));

var _format = _interopRequireDefault(require("date-fns/format"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Day = function Day(_ref) {
  var blockClassName = _ref.blockClassName,
      date = _ref.date,
      handleOnClick = _ref.handleOnClick,
      handleOnEnter = _ref.handleOnEnter,
      isCurrentMonth = _ref.isCurrentMonth,
      isDisabled = _ref.isDisabled,
      isHighlighted = _ref.isHighlighted,
      isMonthNext = _ref.isMonthNext,
      isMonthPrev = _ref.isMonthPrev,
      isNonSelectable = _ref.isNonSelectable,
      isSelectable = _ref.isSelectable,
      isSelected = _ref.isSelected,
      isSelectionEnd = _ref.isSelectionEnd,
      isSelectionStart = _ref.isSelectionStart,
      isToday = _ref.isToday,
      isWeekend = _ref.isWeekend,
      isWorkday = _ref.isWorkday;
  return _react.default.createElement("button", {
    className: (0, _classnames.default)("".concat(blockClassName, "-day"), {
      'is-current_month': isCurrentMonth,
      'is-disabled': isDisabled,
      'is-end_selection': isSelectionEnd,
      'is-highlighted': isHighlighted,
      'is-next_month': isMonthNext,
      'is-not_selectable': isNonSelectable,
      'is-prev_month': isMonthPrev,
      'is-selectable': isSelectable,
      'is-selected': isSelected,
      'is-start_selection': isSelectionStart,
      'is-today': isToday,
      'is-weekend': isWeekend,
      'is-working_day': isWorkday
    }),
    onClick: handleOnClick,
    onMouseEnter: handleOnEnter,
    value: date
  }, (0, _format.default)(date, 'D'));
};

Day.propTypes = {
  blockClassName: _propTypes.default.string.isRequired,
  date: _propTypes.default.string.isRequired,
  handleOnClick: _propTypes.default.func,
  handleOnEnter: _propTypes.default.func,
  isCurrentMonth: _propTypes.default.bool.isRequired,
  isDisabled: _propTypes.default.bool.isRequired,
  isHighlighted: _propTypes.default.bool.isRequired,
  isMonthNext: _propTypes.default.bool.isRequired,
  isMonthPrev: _propTypes.default.bool.isRequired,
  isNonSelectable: _propTypes.default.bool.isRequired,
  isSelectable: _propTypes.default.bool.isRequired,
  isSelected: _propTypes.default.bool.isRequired,
  isSelectionEnd: _propTypes.default.bool.isRequired,
  isSelectionStart: _propTypes.default.bool.isRequired,
  isToday: _propTypes.default.bool.isRequired,
  isWeekend: _propTypes.default.bool.isRequired,
  isWorkday: _propTypes.default.bool.isRequired
};
var _default = Day;
exports.default = _default;
